Source/WebKit/Shared/Extensions/WebExtensionUtilities.cpp

```diff
@@ -98,6 +99,108 @@ RefPtr<JSON::Object> mergeJSON(RefPtr<JSON::Object> jsonA, RefPtr<JSON::Object>
     return mergedObject;
 }
 
+WTF_ATTRIBUTE_PRINTF(1, 0)
+static String formatString(const char* format, va_list arguments)
+{
+    va_list args;
+    va_copy(args, arguments);
+
+ALLOW_NONLITERAL_FORMAT_BEGIN
+
+#if PLATFORM(COCOA)
+    auto cfFormat = adoptCF(CFStringCreateWithCStringNoCopy(kCFAllocatorDefault, format, kCFStringEncodingUTF8, kCFAllocatorNull));
+    auto cfResult = adoptCF(CFStringCreateWithFormatAndArguments(0, 0, cfFormat.get(), args));
+    va_end(args);
+    return cfResult.get();
+#endif
+
+#if PLATFORM(WIN)
+    int len = _vscwprintf(format, args);
+    Vector<wchar_t> buffer(len + 1);
+    _vsnwprintf(buffer.data(), len + 1, format, args);
+    va_end(args);
+    return { buffer.data() };
+#else
+    char ch;
+    int result = vsnprintf(&ch, 1, format, args);
+#endif
```

Comment:
This should go all the way to the end of the function. It is dead code for the other platforms otherwise.